### DISCUSSION TITLE
How to access the pytest.mark.xfail reason keyword argument?

### DISCUSSION THREAD

**Discussion by adehad** [pytest-dev/pytest-html#467] [Q&A]
I've just migrated a project from Py27 using `pytest-html==1.22.1` and am now using `pytest-html==3.1.1`, I have a plugin that was previously able to get the reason keyword argument using the `wasxfail` attribute of `report`, but now this seems to be a boolean, which is understandable.

Is there a suggestion on how to access this `reason` in a clean way?

This should be a minimal working example, (stripping the other logic my plugin does hopefully):

```python
@pytest.hookimpl(optionalhook=True)
def pytest_html_results_table_header(cells):
    """Change the report table headers."""
    # By default the headers stored in cells are RESULT, TEST, DURATION, LINKS
    result, test, duration, _ = cells
    cells[:] = [
        test,
        result,
        html.th("Manual Tests:"),
        duration,
    ]


def get_manual_tests(item):
    return "\n".join(
        [i.kwargs.get("reason", "") for i in item.own_markers if i.name == "xfail"]
    )


@pytest.hookimpl(optionalhook=True)
def pytest_html_results_table_row(report, cells):
    """Change the report table data."""
    result, test, duration, _ = cells
    cells[:] = [
        test,
        result,
        html.td(report.manual_tests),
        duration,
    ]


@pytest.hookimpl(hookwrapper=True)
def pytest_runtest_makereport(item, call):
    outcome = yield
    report = outcome.get_result()
    report.manual_tests = get_manual_tests(item)

```

and say have a test file like this :

```python
manual_test = partial(pytest.mark.xfail, strict=False)

class TestCoreFunctions(object):
    @manual_test(reason="T-1, T-2"))
    def test_ui(self):
        pass

```

**Comment by jeffwright13** [comment]
I am running Pytest 7.1.1 with Python 3.9.9.

`wasxail` is a string that gives the reason. It's not a boolean. You can see this as it gets populated by looking at the Pytest source code, file `skipping.py`, in function `pytest_runtest_makereport`. However, `wasxffail` is not visible in every TestReport (which is what your `report` variable is). `wasxfail` only gets populated when a test is actually marked xfail, and is only available during the `call` phase.

How do you develop your plugin? Do you have the ability to set breakpoints in your IDE as your test runs proceed? If so, perhaps you can use that to examine Pytest's internals while running your tests.

That said I'm pretty sure the `reason` variable should be available for failed tests, but when I searched the Pytest source code I came up with >1000 matches, too many to examine right now!
